pkg/oci/remote/write.go

```diff
@@ -91,6 +92,109 @@ func WriteSignedImageIndexImages(ref name.Reference, sii oci.SignedImageIndex, o
 	return nil
 }
 
+// WriteSignedImageIndexImagesBulk writes the images within the image index.
+// Bulk version.  Uses targetRegistry for multiple images/sigs/atts.
+// This includes the signed image and associated signatures in the image index
+func WriteSignedImageIndexImagesBulk(targetRegistry string, sii oci.SignedImageIndex, opts ...Option) error {
```

Comment:
I wonder if we could refactor the [copy command](https://github.com/sigstore/cosign/blob/main/cmd/cosign/cli/copy/copy.go#L41) here, which will handle copying over the image and any signatures/attestations associated with it, rather than having to do it manually here.

Right now the `copy` takes a source image and a destination image as a string. We could look into pulling out the copy logic into a reusable function which takes a `SignedEntity` and passing through to that here?

Reply:
Good thoughts.  Let me look into this more.

Reply:
My only concern for the copy command logic is the usabiltiy in an air-gapped situation where cosign may not be able to reach the source image in its original registry/repo.  Will the `copy` mechanics work if the `SignedEnitity` is what's on local disk via `cosign save`?